cmd/vorteil/virtualizers.go

```diff
@@ -10,20 +10,41 @@ import (
 	"path/filepath"
 	"runtime"
 
+	"github.com/beeker1121/goque"
 	"github.com/thanhpk/randstr"
 	"github.com/vorteil/vorteil/pkg/ext"
 	"github.com/vorteil/vorteil/pkg/vcfg"
 	"github.com/vorteil/vorteil/pkg/vdisk"
 	"github.com/vorteil/vorteil/pkg/vimg"
 	"github.com/vorteil/vorteil/pkg/virtualizers/firecracker"
 	"github.com/vorteil/vorteil/pkg/virtualizers/hyperv"
+	"github.com/vorteil/vorteil/pkg/virtualizers/iputil"
 	"github.com/vorteil/vorteil/pkg/virtualizers/qemu"
 	"github.com/vorteil/vorteil/pkg/virtualizers/virtualbox"
 	"github.com/vorteil/vorteil/pkg/vpkg"
 )
 
+var ips *goque.Queue
+
 // buildFirecracker does the same thing as vdisk.Build but it returns me a calver of the kernel being used
 func buildFirecracker(ctx context.Context, w io.WriteSeeker, cfg *vcfg.VCFG, args *vdisk.BuildArgs) (string, error) {
+	var err error
+	for i := range cfg.Networks {
+		if ips == nil {
+			ips, err = iputil.NewIPStack()
+			if err != nil {
+				return "", err
+			}
+		}
+		ip, err := ips.Dequeue()
+		if err != nil {
```

Comment:
ips might potentially never close if there is an error here.You should be able to fix this by moving
`ips, err = iputil.NewIPStack()` & `defer ips.Close()` before the for loop.